src/XMLimport.cpp

```diff
@@ -96,184 +96,158 @@ bool XMLimport::importPackage( QIODevice * device, QString packName, int moduleF
+            if( name() == "MudletPackage" ) {
                 readPackage();
             }
-            else if( name() == "map" )
-            {
+            else if( name() == "map" ) {
                 readMap();
                 mpHost->mpMap->audit();
             }
-            else
-            {
-                qDebug()<<"ERROR:name="<<name().toString()<<"text:"<<text().toString();
+            else {
+                qDebug().nospace() << "XMLimport::importPackage(...) ERROR: unrecognised element with name: "
+                                   << name().toString()
+                                   << " and content: "
+                                   << text().toString();
             }
         }
     }
 
-    if( gotTimer && ! packName.isEmpty() )
-    {
-        mpTimer->setIsActive( true );
-        mpTimer->enableTimer( mpTimer->getID() );
-    }
-
-    if( gotAlias && ! packName.isEmpty() )
-    {
-        mpAlias->setIsActive( true );
-    }
 
-    if( gotAction && ! packName.isEmpty() )
-    {
-        mpHost->getActionUnit()->updateToolbar();
-    }
+    if( ! packName.isEmpty() ) {
+        if( ! gotTrigger ) {
+             mpHost->getTriggerUnit()->unregisterTrigger( mpTrigger );
+             delete mpTrigger;
+        }
 
-    if( ! packName.isEmpty())
-    {
-       if( ! gotTrigger )
-       {
-            mpHost->getTriggerUnit()->unregisterTrigger( mpTrigger );
-            delete mpTrigger;
-       }
-       if( ! gotTimer )
-       {
+        if( gotTimer ) { // packName is NOT empty for modules...!
+            mpTimer->setIsActive( true );
+            mpTimer->enableTimer( mpTimer->getID() );
+        }
+        else {
             mpHost->getTimerUnit()->unregisterTimer( mpTimer );
             delete mpTimer;
-       }
-       if( ! gotAlias )
-       {
+        }
+
+        if( gotAlias ) {
+            mpAlias->setIsActive( true );
+        }
+        else {
             mpHost->getAliasUnit()->unregisterAlias( mpAlias );
             delete mpAlias;
-       }
-       if( ! gotAction )
-       {
+        }
+
+        if( gotAction ) {
+            mpHost->getActionUnit()->updateToolbar();
+        }
+        else {
             mpHost->getActionUnit()->unregisterAction( mpAction );
             delete mpAction;
-       }
-       if( ! gotKey )
-       {
-            mpHost->getKeyUnit()->unregisterKey( mpKey );
-            delete mpKey;
-       }
-       if( ! gotScript )
-       {
-            mpHost->getScriptUnit()->unregisterScript( mpScript );
-            delete mpScript;
-       }
+        }
+
+        if( ! gotKey ) {
+             mpHost->getKeyUnit()->unregisterKey( mpKey );
+             delete mpKey;
+        }
+
+        if( ! gotScript ) {
+             mpHost->getScriptUnit()->unregisterScript( mpScript );
+             delete mpScript;
+        }
     }
+
     return ! error();
 }
 
 void XMLimport::readVariableGroup( TVar *pParent )
 {
-    TVar * var;
-    if( pParent )
```

Comment:
This construct is used for each of the Mudlet Item types but it is pointless and refactors down to the single line I've used.